src/Toml/Bi/Combinators.hs

```diff
@@ -162,28 +200,22 @@ day = match _Day
 timeOfDay :: Key -> TomlCodec TimeOfDay
 timeOfDay = match _TimeOfDay
 
--- TODO: implement using bijectionMaker
--- | Parser for array of values. Takes converter for single array element and
+-- | Parser for list of values. Takes converter for single array element and
 -- returns list of values.
-arrayOf :: forall a . Typeable a => BiMap a AnyValue -> Key -> TomlCodec [a]
-arrayOf bimap key = Codec input output
-  where
-    input :: Env [a]
-    input = do
-        mVal <- asks $ HashMap.lookup key . tomlPairs
-        case mVal of
-            Nothing -> throwError $ KeyNotFound key
-            Just (AnyValue (Array arr)) -> case arr of
-                []      -> pure []
-                l@(x:_) -> case mapM (matchValueBackward bimap) l of
-                    Nothing   -> throwError $ TypeMismatch key (typeName @a) (valueType x)
-                    Just vals -> pure vals
-            Just _ -> throwError $ TypeMismatch key (typeName @a) TArray
-
-    output :: [a] -> St [a]
-    output a = do
-        anyVal <- MaybeT $ pure $ forward (_Array bimap) a
-        a <$ modify (\(TOML vals tables) -> TOML (HashMap.insert key anyVal vals) tables)
+arrayOf :: Typeable a => BiMap a AnyValue -> Key -> TomlCodec [a]
+arrayOf bimap = match (_Array bimap)
+
+setOf :: (Typeable a, Ord a) => BiMap a AnyValue -> Key -> TomlCodec (Set a)
+setOf bimap = match (_Set bimap)
```

Comment:
Let's call all such converters as `arraySetOf` to explicitly tell that they will use use arrays from TOML. Once we have array of tables support, we can then introduce `setOf` and `listOf` and others.